.github/workflows/generate.yml

```diff
@@ -0,0 +1,47 @@
+name: Generate
+
+on:
+  pull_request:
+    branches: [ master ]
+
+jobs:
+  generate:
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v2
+    - uses: actions/cache@v2
+      id: generate-dep-cache
+      with:
+        path: ~/go/pkg/mod
+        key: ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
+    - name: Fetch Dependencies
+      run: go get ./...
+    -
+      name: Install Protoc
+      uses: arduino/setup-protoc@v1.1.0
+      with:
+        version: '3.11.4'
+    -
+      name: Install Protoc-gen-go
+      run: |
+        go get github.com/golang/protobuf/protoc-gen-go@v1.4.2
+        go get google.golang.org/grpc/cmd/protoc-gen-go-grpc@v0.0.0-20200617041141-9a465503579e
+    - name: Generate
+      run: go generate ./...
+    - name: Check
+      # git sttatus --porcelain shows changed files.
+      # the echoed output provides github action output format, per
+      # https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjobs_idoutputs
```

Comment:
Ok got it, we re-generate the protobuf code with the version of `protoc-gen-go` we expect to be using and if it causes any changes in the code then someone has generated them with a different version.

I was was expecting something more like this ^^^

Reply:
the action is set up as a general re-generation of generated files, and isn't protoc specific though